File: app/src/main/java/com/mapmyindia/sdk/demo/java/activity/NearbyPluginActivity.java
package com.mapmyindia.sdk.demo.java.activity;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.mapmyindia.sdk.demo.R;
import com.mapmyindia.sdk.demo.databinding.ActivityNearbyPluginBinding;
import com.mapmyindia.sdk.nearby.plugin.MapmyIndiaNearbyFragment;

public class NearbyPluginActivity extends AppCompatActivity {

    private ActivityNearbyPluginBinding mBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_nearby_plugin);

        MapmyIndiaNearbyFragment nearbyFragment = MapmyIndiaNearbyFragment.newInstance();

        getSupportFragmentManager().
                beginTransaction().
                replace(mBinding.fragmentContainer.getId(), nearbyFragment, MapmyIndiaNearbyFragment.class.getSimpleName())
                .commit();


        nearbyFragment.setMapmyIndiaNearbyCallback(nearbyAtlasResult -> {
            // getSupportFragmentManager().beginTransaction().remove(nearbyFragment).commit();

            Toast.makeText(NearbyPluginActivity.this, nearbyAtlasResult.placeAddress, Toast.LENGTH_SHORT).show();
        });
    }
}

File: app/src/main/java/com/mapmyindia/sdk/demo/kotlin/activity/NearbyPluginActivity.kt
package com.mapmyindia.sdk.demo.kotlin.activity

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.mapmyindia.sdk.demo.R
import com.mapmyindia.sdk.demo.databinding.ActivityNearbyPluginBinding
import com.mapmyindia.sdk.nearby.plugin.IMapmyIndiaNearbyCallback
import com.mapmyindia.sdk.nearby.plugin.MapmyIndiaNearbyFragment
import com.mmi.services.api.nearby.model.NearbyAtlasResult

class NearbyPluginActivity : AppCompatActivity() {

    private lateinit var mBinding: ActivityNearbyPluginBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_nearby_plugin)
        val nearbyFragment = MapmyIndiaNearbyFragment.newInstance()
        supportFragmentManager.beginTransaction().add(mBinding.fragmentContainer.id, nearbyFragment, MapmyIndiaNearbyFragment::class.java.simpleName)
                .commit()

        nearbyFragment.setMapmyIndiaNearbyCallback(object : IMapmyIndiaNearbyCallback {
            override fun getNearbyCallback(nearbyAtlasResult: NearbyAtlasResult) {
                Toast.makeText(this@NearbyPluginActivity, nearbyAtlasResult.placeAddress, Toast.LENGTH_SHORT).show()
            }
        })
    }
}